code/__HELPERS/mobs.dm

```diff
@@ -231,13 +231,21 @@ GLOBAL_LIST_EMPTY(species_list)
 		checked_health["health"] = health
 	return ..()
 
+///Checks to see if our mob is currently in a do_after with the target as the target
+/mob/proc/is_interacting_with(atom/target)
+	return(target in do_afters)
+
 /proc/do_after(mob/user, var/delay, needhand = 1, atom/target = null, progress = 1, datum/callback/extra_checks = null)
 	if(!user)
 		return 0
 	var/atom/Tloc = null
 	if(target && !isturf(target))
 		Tloc = target.loc
 
+	if(target)
+		LAZYADD(user.do_afters, target) // i know technically it already handles if we try to add a null but we're already checking if target exists so shut up
+		LAZYADD(target.targeted_by, user) // i know technically it already handles if we try to add a null but we're already checking if target exists so shut up

```

Comment:
No need for those comments. We don't want to at null entries to a list, nor we want to runtime. The code is fine.